src/helpers/routeHelper.js

```diff
@@ -1,47 +1,131 @@
+
+  static async getCities(req, res) {
+    const cities = await CityController.getCities();
+    ExpressMiddleware.sendResponse(res, 200, cities);
+  }
+
+  static async updateCity(req, res) {
+    await CityController.updateCity(
+      req.params.id, req.body.name, req.body.country,
+      req.body.capital, req.body.lat, req.body.long,
+    );
+    const city = await CityController.getCity(req.params.id);
+    ExpressMiddleware.sendResponse(res, 200, city);
+  }
+
+  static async deleteCity(req, res) {
+    const city = await CityController.deleteCity(req.params.id);
+    ResponseHandler.mongoResponse(res, city.result, 'city', req.params.id);
+  }
+}
+
+export class CredentialsRouteUtils {
+  static async verifyCredentials(req, res) {
+    const creds = await CredentialsController.verifyCredentials(req.body.login, req.body.password);
+    if (creds) {
+      const user = await UserController.getUser(creds._id);
+      const data = {
+        user: {
+          username: user.name,
+          email: user.email,
+        },
+        token: JWT.generateJwt(user.toObject()),
+      };
+      ExpressMiddleware.sendResponse(res, 200, data);
     } else {
-      next('route');
+      ExpressMiddleware.sendResponse(res, 403, locale('auth_failure'));
+    }
+  }
+
+  static verifyCredentialsPassport(req, res) {
+    const data = {
+      user: req.user,
+      token: JWT.generateJwt(req.user.toJSON()),
+    };
+    ExpressMiddleware.sendResponse(res, 200, data);
+  }
+
+  static verifyCredentialsPassportSocial(req, res) {
+    const data = req.user;
+    ExpressMiddleware.sendResponse(res, 200, data);
+  }
+}
+
+export class ProductRouteUtils {
+  static async addProduct(req, res) {
+    try {
+      const product = await ProductController.addProduct(
+        req.body._id, req.body.name, req.body.brand,
+        req.body.company, req.body.price, req.body.isbn,
+      );
+      ExpressMiddleware.sendResponse(res, 200, product);
+    } catch (err) {
+      ExpressMiddleware.sendResponse(res, 409, util.format(locale('product_exists'), req.body._id));
     }
   }
 
-  static getProducts(req, res) {
-    const products = ProductController.getProducts();
-    res.json(products);
+  static async getProduct(req, res) {
+    const product = await ProductController.getProduct(req.params.id);
+    if (product) {
+      ExpressMiddleware.sendResponse(res, 200, product);
+    } else {
+      ExpressMiddleware.sendResponse(res, 404, util.format(locale('product_not_found'), req.params.id));
+    }
   }
 
-  static sendProduct(req, res) {
-    res.json(req.product);
+  static async getProducts(req, res) {
+    const products = await ProductController.getProducts();
+    ExpressMiddleware.sendResponse(res, 200, products);
   }
 
-  static productNotFound(req, res) {
-    res.status(404);
-    res.end(`${config.product_not_found}: ${req.params.id}`);
+  static async deleteProduct(req, res) {
+    const product = await ProductController.deleteProduct(req.params.id);
```

Comment:
It'd be good to remove reviews connected with this product.
Another option - make reviews a part of products collection.

Reply:
Fixed